streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractKeyValueStoreTest.java

```diff
@@ -389,4 +389,9 @@ public void shouldNotThrowConcurrentModificationException() {
 
         assertEquals(new KeyValue<>(0, "zero"), results.next());
     }
+
+    @Test
+    public void shouldNotThrowInvalidRangeExceptionWithNegativeFromKey() {
+        store.range(-1, 1);
```

Comment:
You can use `org.apache.kafka.streams.processor.internals.testutil.LogCaptureAppender` to assert the correct log message

Reply:
Ah thanks, will add to tests